src/utils/iServer.ts

```diff
@@ -57,6 +58,7 @@ export abstract class AbstractServer implements IServer {
   }
 
   abstract sendMessage(messageObj: any): Promise<any>;
+  abstract close(): void;
```

Comment:
This `close()` can be non-abstract, so there will be no need in empty implementations in its descendants.